app/ide-desktop/lib/dashboard/src/components/JSONSchemaInput.tsx

```diff
@@ -49,14 +51,10 @@ export default function JSONSchemaInput(props: JSONSchemaInputProps) {
 
   React.useEffect(() => {
     setValue(valueRaw)
-    // `initializing` is not a dependency.
-    // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [valueRaw])
 
   React.useEffect(() => {
-    setValueRaw(value)
-    // `setStateRaw` is a callback, not a dependency.
-    // eslint-disable-next-line react-hooks/exhaustive-deps
+    setValueRawRef.current(value)
```

Comment:
I assume "useEffect" might be too late here. Since we're assigning to a ref, we can safely do this during the render. Please refer to https://react.dev/learn/you-might-not-need-an-effect to learn more about patterns you can use to avoid using "useEffect" 

P.S. We have `useSyncRef` hook(app/ide-desktop/lib/dashboard/src/hooks/syncRefHooks.ts) that does exactly what you want